### PR TITLE
[LLVMGPU] Enable tensor.unpack op e2e execution on cuda.

### PR DESCRIPTION
**Pull request by hanhanW**

### CODE REVIEW

**Comment by hanhanW** [comment]
This depends on https://github.com/openxla/iree/pull/12458, please take a look at the last commit, thanks!

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp [hunk 1]
```diff
@@ -442,6 +442,39 @@ static LogicalResult setPackConfig(func::FuncOp entryPoint,
...
+  for (auto [pos, size] : llvm::zip_equal(dimPos, innerTiles)) {
+    if (tileSizes[pos] == 0 || ShapedType::isDynamic(size)) continue;
+    tileSizes[pos] = llvm::alignTo(tileSizes[pos], size);
+    if (pos >= 3) continue;
+    // Distribute one tile on one thread.
+    if (srcType.isDynamicDim(pos)) {
+      workgroupSizes[pos] = tileSizes[pos] / size;
+    } else {
+      workgroupSizes[pos] =
+          std::min(tileSizes[pos], srcType.getDimSize(pos) * size) / size;
```
Comment (ThomasRaoux) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp]: why is the inner dim position deciding which ids do we map a loop to? To get decent performance we need to try to map the most inner dimensions to the fastest moving ids (so mapping dim N -> id.x N-1 -> id.y.
It sounds like this is not what that would do?

Also could we clearly comment on what part of this logic is needed for functionality if there is any?

Maybe I'm just not understanding this logic correctly and some extra comments are enough.

Comment (hanhanW) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp]: I assumed that the rank of unpacked layout is always less than two. That's most of the cases we've seen, particularly in matmul. Let me generalize the lowering config a bit to handle other cases..

**Comment by hanhanW** [comment]
I can't reproduce it on my workstation. I can reproduce it on a VM that has A100 now. Seems to be race. I'm able to pass the test if I set the first level of tiling sizes small enough, need more investigation..

**Comment by benvanik** [comment]
Closing stale PR.

**Pull request closed without merging**